http-handler/src/main/java/com/flipkart/poseidon/handlers/http/impl/oauth/OAuthTokenGenerator.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2015 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.flipkart.poseidon.handlers.http.impl.oauth;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.flipkart.phantom.task.spi.TaskContext;
+import com.flipkart.poseidon.handlers.http.HttpResponseData;
+import com.flipkart.poseidon.handlers.http.impl.SinglePoolHttpTaskHandler;
+import com.flipkart.poseidon.handlers.http.utils.StringUtils;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.utils.URIBuilder;
+import org.trpr.platform.core.impl.logging.LogFactory;
+import org.trpr.platform.core.spi.logging.Logger;
+
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OAuthTokenGenerator extends SinglePoolHttpTaskHandler {
```

Comment:
Class level comments here as well. That during construction, it gets a token and then periodically refreshes, how it handles errors etc